### PR TITLE
Auto/fix surround obstacle checker

### PR DESCRIPTION
**Pull request by Sugatyon**
## Description

change trajectory to trajectory points(vector of trajectory point) in function

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Read [pull request guidelines][pull-request-guidelines]
- [x] Code follows [coding guidelines][coding-guidelines]
- [ ] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[clang-tidy-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/ClangTidyGuideline/
[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[pull-request-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/PullRequestGuideline/
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: planning/scenario_planning/lane_driving/motion_planning/surround_obstacle_checker/src/node.cpp [hunk 1]
```diff
@@ -102,7 +104,8 @@ void SurroundObstacleCheckerNode::pathCallback(
   }
 
   // get closest idx
-  const size_t closest_idx = getClosestIdx(*input_msg, current_pose);
+  const size_t closest_idx =
+    getClosestIdx(autoware_utils::convertToTrajectoryPointArray(*input_msg), current_pose);
```
Comment (tkimura4) [inline planning/scenario_planning/lane_driving/motion_planning/surround_obstacle_checker/src/node.cpp:108]: output_trajectory_pointsで良い...???

Comment (TakaHoribe) [inline planning/scenario_planning/lane_driving/motion_planning/surround_obstacle_checker/src/node.cpp:108]: ここ `getClosestIdx(output_trajectory_points)`で良いのでは？

Comment (TakaHoribe) [reply planning/scenario_planning/lane_driving/motion_planning/surround_obstacle_checker/src/node.cpp:108]: まけた・・・

Comment (tkimura4) [reply planning/scenario_planning/lane_driving/motion_planning/surround_obstacle_checker/src/node.cpp]: 6abb1b1

Comment (tkimura4) [reply planning/scenario_planning/lane_driving/motion_planning/surround_obstacle_checker/src/node.cpp]: 6abb1b1

**Review by tkimura4 — APPROVED** [review]
LGTM

**Pull request merged**